src/ucp/core/ucp_ep.c

```diff
@@ -2727,6 +2730,17 @@ static void ucp_ep_req_purge(ucp_ep_h ucp_ep, ucp_request_t *req,
         }
 
         ucp_request_put(req);
+    } else if (req->flags & UCP_REQUEST_FLAG_RMA_SW_SENT) {
+        ucs_assert(req->send.ep == ucp_ep);
+
+        if (ucp_ep->worker->context->config.ext.proto_enable) {
```

Comment:
proto_enable mode purge should work differently
so for now can assert/avoid purge if ucp_ep->worker->context->config.ext.proto_enable is true